python/cuml/test/experimental/test_explainer_shap.py

```diff
@@ -0,0 +1,177 @@
+    (10, cuml.LinearRegression): [-3.47197726, -12.13657959, -43.05540892,
+                                  -41.44955195, -4.1909009, -30.91657623,
+                                  -14.73675613, 23.92447365, 15.73265123,
+                                  -45.94585396],
+    (4, cuml.KNeighborsRegressor): [58.13167305, 139.33765425, 28.08136872,
+                                    13.12541971],
+    (10, cuml.KNeighborsRegressor): [-3.47197726, -12.13657959, -43.05540892,
+                                     -41.44955195, -4.1909009, -30.91657623,
+                                     -14.73675613, 23.92447365, 15.73265123,
+                                     -45.94585396]
+}
+
+
+# todo: use tags to generate the correct dataset
+@pytest.fixture(scope="session")
+def single_dataset():
+    X, y = make_classification(100, 5, random_state=42)
+    X = X.astype(np.float32)
+    y = y.astype(np.float32)
+    return X, y
+
+
+def func_positional_arg(func):
+    if hasattr(func, "__code__"):
+        all_args = func.__code__.co_argcount
+        if func.__defaults__ is not None:
+            kwargs = len(func.__defaults__)
+        else:
+            kwargs = 0
+        return all_args - kwargs
+    return 2
+
+
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+@pytest.mark.parametrize("nfeatures", [4, 10])
+@pytest.mark.parametrize("nbackground", [80])
+@pytest.mark.parametrize("model", [cuml.LinearRegression,
+                                   cuml.KNeighborsRegressor])
+def test_kernel_shap_standalone(dtype, nfeatures, nbackground, model):
+    X, y = cuml.datasets.make_regression(n_samples=nbackground + 1,
+                                         n_features=nfeatures,
+                                         noise=0.1,
+                                         random_state=42)
+
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, test_size=1, random_state=42)
+
+    X_train = X_train.astype(np.float32)
+    X_test = X_test.astype(np.float32)
+    y_train = y_train.astype(np.float32)
+    y_test = y_test.astype(np.float32)
+
+    mod = model().fit(X_train, y_train)
+
+    cu_explainer = cuml.experimental.explainer.KernelSHAP(model=mod.predict,
+                                                          data=X_train,
+                                                          gpu_model=True)
+
+    cu_shap_values = cu_explainer.shap_values(X_test[0])
+
+    assert array_equal(cu_shap_values, golden_results[nfeatures, model],
+                       1e-1, with_sign=True)
+
+
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+@pytest.mark.parametrize("nfeatures", [4, 100])
+@pytest.mark.parametrize("nbackground", [10, 80])
+@pytest.mark.parametrize("model", [cuml.LinearRegression,
+                                   cuml.KNeighborsRegressor])
+def test_kernel_gpu_cpu_shap(dtype, nfeatures, nbackground, model):
+    if not has_shap():
+        pytest.skip("Need SHAP installed for these tests")
```

Comment:
should we add shap to ci/testing?

Reply:
Yes, but I was thinking to add it for early 0.18, not a huge fan of modifying the CI testing environment this late in the cycle